### DISCUSSION TITLE
[Unknown Mod][Adventure Map] Creature recruit window, add a "Marketplace" button

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#5508] [Ideas]
Add a "Marketplace" button (only if not enough Gold) in the Creature recruit window.

![Marketplace button fh2 01](https://user-images.githubusercontent.com/43583994/116811699-d6867e00-ab4a-11eb-9e59-46386fdd9fcc.png)

In fheroes2:

![Creature recruit window 01](https://user-images.githubusercontent.com/43583994/172149937-4a9a0a91-fc38-4b66-83e0-ba0ed821a9f1.png)

This is not an option.
This can be implemented directly in fheroes2.

**Comment by oleg-derevenetz** [comment]
The Marketplace button is added to standard dialog boxes in cases when there is not enough money for an action, and either the Marketplace is currently unavailable (for example, during a battle - in the surrender dialog), or this action cannot be repeated later (for example, when wandering units are willing to join your army for a cost - you should make a decision immediately). However, if you can simply repeat the hiring action later using the space bar button after visiting a Marketplace, adding this button means just cluttering up the interface IMHO. We should avoid cluttering the interface with various frills without much need.

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz,

> The Marketplace button is added to standard dialog boxes in cases when there is not enough money for an action, and either the Marketplace is currently unavailable (for example, during a battle - in the surrender dialog), or this action cannot be repeated later (for example, when wandering units are willing to join your army for a cost - you should make a decision immediately). However, if you can simply repeat the hiring action later using the space bar button after visiting a Marketplace, adding this button means just cluttering up the interface IMHO. We should avoid cluttering the interface with various frills without much need.

Let's see the actions to be performed to recruit medusas from the original post image, with and without the "Marketplace" button.

Without the "Marketplace" button:
- Press the "CANCEL" button to close the "Recruit Medusas" window.
- Double click on the icon of one of the castles in the castle list.
- Possibly change castle, if the marketplace is not built in the first selected castle.
- Click on the marketplace graphic to open the "Marketplace" window.
- Make the trade to get the requested resources.
- Press the "EXIT" button to close the "Marketplace" window.
- Press the "EXIT" button to close the castle screen.
- Select the hero at the location of the ruins.
- Press the space bar to open the "Ruins" window.
- Press the "YES" button to open the "Recruit Medusas" window.
- Proceed with the recruitment of medusas.
- Press the "OKAY" button to confirm the recruitment of medusas.

With the "Marketplace" button:
- Press the "Marketplace" button.
- Make the trade to get the requested resources.
- Press the "EXIT" button to close the "Marketplace" window.
- Proceed with the recruitment of medusas.
- Press the "OKAY" button to confirm the recruitment of medusas.

As you can see, there would be a real improvement in the use of the interface by making it much more unser friendly. :)

**Reply by oleg-derevenetz** [reply]
> As you can see, there would be a real improvement in the use of the interface by making it much more unser friendly. :)

We need to stop somewhere in the process of turning the interface "more user friendly", otherwise it will turn into adding everywhere shortcuts to everything.

**Reply by LeHerosInconnu** [reply]
@oleg-derevenetz.

> > As you can see, there would be a real improvement in the use of the interface by making it much more unser friendly. :)
> 
> We need to stop somewhere in the process of turning the interface "more user friendly", otherwise it will turn into adding everywhere shortcuts to everything.

I think in this case the button should be added, especially since players are used to seeing it available in other windows of the game, and it would be very convenient to have it.

**Reply by zenseii** [reply]
@oleg-derevenetz has a valid point and we should consider this for other cases where we want to add buttons. Generally I think it is a good idea to add these buttons if there is no way to exit and return to the dialog we are in, like for creatures joining, surrendering in battle, choosing skill when leveling up etc. Since for creature recruitment we can come back again, maybe it would be better to adopt something like pressing 'm' to pull up the marketplace when on the adventure map, provided that you own one? This would be in line with how it's done in other Heroes entries.

This would need an interface solution same as for sleep which also has no button or way to access at the moment for players without a keyboard or gamepad.

**Reply by LeHerosInconnu** [reply]
Hello @oleg-derevenetz and @zenseii,

Initially, I had the same thought as both of you. :)
"The player can use the space key on the keyboard after getting the missing resources to come back and buy medusas."
But after being confronted with this way of working, I thought it would be a shame to force the user to do all these actions when the button and the code that goes with it already exist.
And also that the user would wonder why this button is not present when he does not have the necessary resources while it is available in other windows.

@oleg-derevenetz.

Rather, I think we should find a solution to put a hero to sleep with only the use of the mouse, even if this action of putting a hero to sleep is little used (in my case).
To stay in the spirit that all actions should be possible to perform using only the mouse.
I don't really like actions that are only possible with "hidden" keyboard shortcuts.
Use of mouse and keyboard should be complementary and not exclusive.
And with a visual interface, the mouse should always be prioritized and available over using the keyboard.

**Reply by zenseii** [reply]
To summarize, we are discussing a way to access the marketplace from the adventure map. This button or option should be simple and fit smoothly in with the rest of the UI.

Additonally this solution should be able tp include other options that are currently hidden behind keyboard keys, like putting heroes in sleep mode.

I don't have any ideas for how to accomplish this at this point in time.

**Reply by ihhub** [reply]
I think for such cases we have to highlight a **Space** button to be used by players as essentially we are adding just a shortcut and complexity of the code. Also take a note that for such windows we need to add a hot key and change the location of buttons.

I am not saying a solid no but I would suggest not to do this right now.
